vault/cors.go

```diff
@@ -0,0 +1,137 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+type CORSConfig struct {
+	enabled          bool
+	allowedOrigins   *[]string
+	allowedHeaders   *map[string]string
+	allowedMethods   *[]string
+	allowCredentials bool
+}
+
+func newCORSConfig() *CORSConfig {
+	defaultOrigins := &[]string{}
+	return &CORSConfig{
+		enabled:        false,
+		allowedOrigins: defaultOrigins,
+		allowedHeaders: &map[string]string{
```

Comment:
Yeah, in looking at this I'm not sure why I coded things this way. Yes, I want the list of allowedOrigins to be nil until configured, but not sure why I put defaultOrigins in there. I'm going to refactor this.